### ISSUE TITLE
JUMPBUG 

### ISSUE THREAD

**Issue by mimphes** [danielkrupinski/Osiris#859]
ok so i finally got fully working jumobug code.
if someone were to add it to the cheat then send it to me it would be greatly appreciated.
the format of code i have is lua. this is skeets jumpbug lua

local get_prop, get_local_player, get, set, vis, trace_line, bit_band = entity.get_prop, entity.get_local_player, ui.get, ui.set, ui.set_visible, client.trace_line, bit.band
local enabled = ui.new_checkbox("MISC", "Miscellaneous", "Jumpbug")
local hotkey = ui.new_hotkey("MISC", "Miscellaneous", "Jumpbug", true)
local infduck = ui.reference("AA", "Other", "Infinite duck")
local max_radias = math.pi * 2
local step = max_radias / 24
local unduck = false

local function jumpbug(event)
    if get(infduck) then
        vis(enabled, true)
        vis(hotkey, true)
        if not get(enabled) then return end
        if not get(hotkey) then return end
        if unduck then
            event.in_duck = 0
            unduck = false
        end
        local player = get_local_player()
        local flags = get_prop(player, "m_fFlags")
        if (bit_band(flags, 1)==1) then return end
        local x, y, z = get_prop(player, "m_vecOrigin")
        for a = 0, max_radias, step do
            local ptX, ptY, ptZ
            ptX = (17 * math.cos( a ) ) + x
            ptY = (17 * math.sin( a ) ) + y
            ptZ = z
            local trace, trace_entity = trace_line(player, ptX, ptY, ptZ, ptX, ptY, ptZ-6)
            if trace~=0 then 
                if trace~=1 then
                    event.in_duck = 1
                    unduck = true
                    break
                end
            end
        end
    else
        set(enabled, false)
        vis(enabled, false)
        vis(hotkey, false)
    end
end
client.set_event_callback("setup_command", jumpbug)

**Comment by tirziz** [comment]
So... If it fully working then you may pull it plz

**Comment by VuBrian22** [comment]
It's in the LUA format Osiris doesn't use LUA. You'll have to convert it to the Osiris SDK for it to work. Did you?

**Comment by thomas0aotian** [comment]
Sure I know sk's lua api

**Comment by mimphes** [comment]
> Sure I know sk's lua api

so can you do it

**Comment by VuBrian22** [comment]
> > Sure I know sk's lua api
> 
> so can you do it

He knows the API. That doesn't mean he knows how to convert the LUA through the Osiris SDK to make it work. He can try though. I can do it myself, however I work as a full time intern as a Cyber Security Threat Analysis so I rarely have time to do so.

**Comment by thomas0aotian** [comment]
@thomas0aotian i was being sacarstic I don't know sk's lua api if I do I can do sdk translation.

**Comment by VuBrian22** [comment]
@thomas0aotian Well I kinda knew that from the start regarding the fact that you said these kinds of things in the past within these forums.

**Comment by thomas0aotian** [comment]
@VuBrian22 emmm:stuck_out_tongue_closed_eyes:
Just kd for the meme

**Comment by VuBrian22** [comment]
> @VuBrian22 emmm😝
> Just kd for the meme

To put it blatantly, (I was trying to nice and respectful in my comment above), Don't say anything if you don't know anything about it. Don't pretend like you know what to do and lead others to believe something that is false. But now you know so we're good! ;)

**Comment by thomas0aotian** [comment]
@VuBrian22 emmm I actually knows what to do lmao.

**Comment by thomas0aotian** [comment]
@VuBrian22 I don't code for living but I at least can translate different sdk

**Comment by VuBrian22** [comment]
@thomas0aotian Well I'm just saying, that's all.

**Comment by xAkiraMiura** [comment]
1: its not hard to translate code from LUA to C++
2: please put your code in code tags so we can see indentation and other stuff
^ lua is hard to read without indentation
3: why are you pasting
4: we have 10,000 other issues like this why do we need another

**Comment by VuBrian22** [comment]
@xAkiraMiura ik however, I'm referring to his past. For example: https://github.com/danielkrupinski/Osiris/issues/808#issuecomment-561293251

**Comment by xAkiraMiura** [comment]
this is the same person? 
also could we get daniel to make issue guidelines because the issues are a terrible mess right now with pasters and lazy people who nothing about coding trying to get people to do work for them.

**Comment by VuBrian22** [comment]
@xAkiraMiura Yup that's pretty much sums up the majority of the people here in the Issues tab. Thanks for your input on it.
